src/waypoint.jsx

```diff
@@ -158,6 +202,10 @@ export default class Waypoint extends React.Component {
     }, 0);
   }
 
+  componentWillReceiveProps(nextProps) {
+    ensureChildrenIsSingleDOMElement(nextProps.children);
```

Comment:
I wonder if it would be worth only calling this if `this.props.children !== nextProps.children`